modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java

```diff
@@ -130,6 +135,49 @@ public BuiltinJsonSchema() {
         return rootNode;
     }
 
+    public @NotNull JsonNode getJsonSchema(final @NotNull BuiltinDataType builtinDataType,
+                                           final @NotNull UInteger[] dimensions) {
+
+        final ObjectNode rootNode = OBJECT_MAPPER.createObjectNode();
+        final ObjectNode propertiesNode = OBJECT_MAPPER.createObjectNode();
+        final ObjectNode valueNode = OBJECT_MAPPER.createObjectNode();
+        rootNode.set("$schema", new TextNode("https://json-schema.org/draft/2019-09/schema"));
+        rootNode.set("title", new TextNode("Array of " + builtinDataType.name() + " JsonSchema"));
+        rootNode.set("type", new TextNode("object"));
+        rootNode.set("properties", propertiesNode);
+        propertiesNode.set("value", valueNode);
+        populatePropertiesForArray(valueNode, builtinDataType, OBJECT_MAPPER, dimensions);
+
+        final ArrayNode requiredAttributes = OBJECT_MAPPER.createArrayNode();
+        requiredAttributes.add("value");
+        rootNode.set("required", requiredAttributes);
+        return rootNode;
+    }
+
+    public static void populatePropertiesForArray(final @NotNull ObjectNode propertiesNode,
+                                                  final @NotNull BuiltinDataType builtinDataType,
+                                                  final @NotNull ObjectMapper objectMapper,
+                                                  final @NotNull UInteger[] dimensions) {
+            final long maxSize = dimensions[0].longValue();
```

Comment:
safety check on the dimensions.lemgth would be nice.

Reply:
Happens in JsonSchemaGenerator before this code is invoked.

Reply:
I know but the method should assert this. It is not part of the method contract.

Reply:
Added a check